Add support for Lightning Address payments and verification

Aniket-pd opened the pull request:
## Summary

This work depends on the backend changes in  
https://github.com/fedimint/fedimint-sdk/pull/225,  
which introduce the two new LN RPCs: `verify_lightning_address` and `pay_lightning_address`.

This PR adds full support for working with Lightning Addresses.  
It includes everything needed for the app to verify a Lightning Address, fetch its details, and send a payment to it.

## What’s Done

- Added new RPC handlers and request types so the LN client can verify or pay Lightning Addresses, making these features available to WASM users.
- Added structured payment details in RPC responses — invoice, limits, comments, and success actions.
- Implemented full LNURL Lightning Address resolution, including metadata extraction and validation of amount and comment.  
  The same logic is now reused in both verification and payment flows.

Aniket-pd committed:
Add support for Lightning Address payments and verification

Introduces methods to verify and pay Lightning Addresses via LNURL, including new request/response types and RPC handlers. Adds invoice detail extraction, validation of amount and comment, and integrates with existing payment flow. This enables clients to interact with Lightning Addresses directly from the client module.

maan2003 commented on modules/fedimint-ln-client/src/lib.rs:
```diff
@@ -1261,6 +1334,106 @@ impl LightningClientModule {
             .await
     }
 
+    async fn request_lightning_address_invoice(
```
yep exactly

Aniket-pd committed:
Make request_lightning_address_invoice public and refactor logic

Changed request_lightning_address_invoice from private to public. Refactored the function to always parse the input as a lightning address, removing support for direct lnurl input. Replaced inline invoice fetching logic with a call to get_invoice.

MrImmortal09 reviewed (changes requested):
I think there’s still a misunderstanding here. You introduced `request_lightning_address_invoice` and `extract_lnurl_description,` but `get_invoice` already handles the same flow. So the functions you need to work on should be structured like this:
	- For handle_rpc → `verify_lightning_address` -> Just create a small function whose only job is to verify the lightning address.
	- For handle_rpc → `pay_lightning_address` -> Create a function that simply calls `get_invoice` to fetch the invoice, and then pay it using `pay_bolt11_invoice.`

This will keep the code cleaner and avoid duplicating logic.

Aniket-pd commented:
Sure, here’s how I’m thinking about the difference:

• verify_lightning_address  
  This is basically a “discovery + validation” helper for Lightning addresses.  
  It resolves the LNURL-pay info, checks limits, comment rules, description,  
  success action, etc., and then fetches the invoice.  
  It returns a richer structure with both the invoice and the metadata so the UI  
  can show everything before the user actually pays.

• get_invoice  
  This is the lower-level, generic utility.  
  It just takes an invoice/LNURL/Lightning address and produces a validated  
  Bolt11 invoice. It doesn’t expose any metadata — only the final invoice.

So in short:  
verify_lightning_address = discovery + metadata + invoice  
get_invoice = invoice only

If you feel this separation doesn’t add much value on the backend, I’m totally fine  
with simplifying and relying only on get_invoice. Just let me know which approach  
you prefer and I’ll update the PR accordingly.

maan2003 commented:
from implementation they look the same https://github.com/fedimint/fedimint/pull/7988/files#diff-d754f9b1f77285e47d11bdecd4e3df652f846e5cedd12ded095ae84816a64cc2R1400-R1408

if they are different (one is not a superset) then two rpcs are fine. I would prefer one RPC.

maan2003 commented on modules/fedimint-ln-client/src/lib.rs:
```diff
@@ -1261,6 +1334,101 @@ impl LightningClientModule {
+            .context("Invalid lightning address")?
+            .lnurl();
+
+        let async_client = lnurl::AsyncClient::from_client(reqwest::Client::new());
```
cache this somewhere and reuse.

Aniket-pd committed:
Remove Lightning Address payment and verification support

Eliminated all code related to Lightning Address payment and verification, including associated request structs, response types, and RPC handlers. LNURL invoice fetching logic is now handled directly in get_invoice, consolidating validation and invoice retrieval.

Aniket-pd committed:
Add get_invoice RPC handler and request struct

Introduces a new 'get_invoice' RPC method to handle invoice retrieval. Adds the GetInvoiceRequest struct for deserializing request parameters.

Aniket-pd committed:
chore(ln-client): add once_cell dependency

Aniket-pd committed:
Replace once_cell with std::sync::LazyLock

elsirion commented:
@maan2003 is this something we still want to do?

maan2003 commented:
we can close, if someone needs this they can reopen.
